service-base/src/main/java/org/eclipse/hono/service/resourcelimits/ResourceLimitChecks.java

```diff
@@ -31,9 +32,26 @@
      *         <p>
      *         The future will be failed with a {@link ServiceInvocationException}
      *         if the check could not be performed.
+     * @deprecated Use {@link #isConnectionLimitReached(TenantObject, SpanContext)} instead.
      */
+    @Deprecated
     Future<Boolean> isConnectionLimitReached(TenantObject tenantObject);
 
+    /**
+     * Checks if the maximum number of connections configured for a tenant
+     * have been reached.
+     *
+     * @param tenantObject The tenant configuration to check the limit against.
+     * @param spanContext The currently active OpenTracing span context that is used to
+     *                    trace the limits verification or {@code null}
+     *                    if no span is currently active.
+     * @return A future indicating the outcome of the check.
+     *         <p>
+     *         The future will be failed with a {@link ServiceInvocationException}
+     *         if the check could not be performed.
+     */
+    Future<Boolean> isConnectionLimitReached(TenantObject tenantObject, SpanContext spanContext);
```

Comment:
what if tenant or span context are `null`?

Reply:
javadoc has been appended with ` @throws NullPointerException if the tenant object is null.`